05-how-to-contribute-to-open-source.md

```diff
@@ -25,139 +29,293 @@ OpenSauced is a powerful tool for finding open source projects to contribute to.
+The best times to update your branches are before you push your changes to the remote repository and while you're waiting for your pull request to be reviewed.
 
-4. **Create a new branch**: Before making any changes, remember to make a new branch in your local repository to work on your contribution. You can create a new branch using the following command:
+### Updating Branches
 
-```
-git checkout -b feature/add-profile-[your-username]
-```
+First, you must update your forked (`origin`) repository:
 
-Replace "your-username" with your GitHub username.
+1. Go to your forked repository on GitHub.
+2. Click the "Sync fork" button.
+3. Click the green "Update branch" button.
 
-5. Run `npm run contributors:add` to add yourself to the contributors list and fill out the form based on your contributions.
+Then, pull the latest changes in the `main` branch in the `origin` repository to your local working branch by following these steps in your terminal:
 
-6. Run `npm run contributors:generate` to generate the new contributors list.
+1. Go to your working branch.
 
-7. Commit your changes and push them to your forked repository.
+   ```bash
+   git checkout YOUR-BRANCH-NAME
+   ```
 
-8. Once pushed, navigate to the original project's repository on GitHub and click the "New pull request" button. In the "base repository" dropdown, select the original repository, and in the "head repository" dropdown, select your forked repository. Choose your branch from the "compare" dropdown, and click "Create pull request". Fill in the required information and submit your pull request.
+2. Pull the latest changes with this command:
 
-Congratulations! You just made an open source contribution! In the next chapter, we'll add this to an open source portfolio: a resume for open source contributions through which you can showcase your work and attract potential employers and friends using OpenSauced. For now, let's wrap up with some notes on onboarding.
+   ```bash
+   git pull origin main
+   ```
 
 ## Merge Conflicts
 
-### Understanding Merge Conflicts
+Merge conflicts are something you'll commonly encounter when contributing to an open source project. When two branches have made different changes to the same line(s) in the same file(s), Git cannot automatically determine which change to keep, resulting in a conflict.
+
+When a merge conflict occurs, Git adds conflict markers (`<<<<<<<`, `=======`, and `>>>>>>`) to indicate the conflicting lines from different branches. Everything between the `<<<<<<<` and `=======` is the changes that you worked on (current changes). And everything between the `=======` to `>>>>>>>` is the incoming changes from the remote `main` branch.
+
+You need to pay attention to the conflicts and decide how you want to resolve them. You can keep only your change, incoming change, or both changes.
+
+### Tips to Avoid Further Conflicts
```

Comment:
It does have a fighting connotation, so you're onto something @adiati98. I came up with another one here: https://github.com/open-sauced/intro/pull/100/files#r1432032740

Reply:
These are not other ways to solve merge conflicts. So this title might not work here.

Reply:
> These are not other ways to solve merge conflicts. So this title might not work here.

Good point @adiati98. I changed my suggestion. Here it is:  https://github.com/open-sauced/intro/pull/100/files#r1432032740

Reply:
Thanks, Chrissy!

But if you see closely, the tips here are not for resolving conflicts. But they are for what contributors need to do to prevent them resolving conflicts over and over again for a PR.

Most problems we see so far when there are conflicts:

- Contributors missed the step to generate the README.
- They didn't complete the PR form.
- It took too long for them to resolve conflicts or completing the PR form, and we've merged in PRs in between.

Still not too happy with this one, but this sounds better.
```suggestion
### Tips to Prevent Resolving Merge Conflicts Repeatedly
```